src/app/infra/repositories/sensors_records_repository.py

```diff
@@ -61,6 +61,15 @@ def get_last_record(self):
 
             if row:  
                 return self.__get_sensors_record(row)
+            
+            else:
```

Comment:
Create a use case called get_last_sensors_record, then you put the verification and the exception that you did in the sensors_records_repository inside the use case, then call the use case inside the view:

```python

from core.use_cases.sensors_records import get_last_sensors_record

def last_sensors_record_page_view():
    last_sensors_record = get_last_sensors_record.execute()

    render_template("template_path", last_record=last_sensors_record)

```

Obs: The sensors_records_repository should return None if there is no record.

Reply:
Why add more complexity to it,since its already working in a simple manner?

Reply:
This will help a lot to make tests. Trust in me

Reply:
Sure mate!,just wondering to help me understand your plan and improve